04-call-summaries/llm.py

```diff
@@ -32,13 +29,14 @@ def ollama_client(
 
     print("LLM settings:", model_name, settings)
     # To connect via another URL: Ollama(base_url='http://localhost:11434', ...)
-    return Ollama(model=model_name, callbacks=callbacks, **settings).invoke(prompt)
+
+    if client:
+        return client.invoke(prompt)
+    return Ollama(model=model_name, callbacks=callbacks, **settings)
```

Comment:
This function can return either a string response or an LLM client. Functions should return the same type of object.